Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessages.h

```diff
@@ -53,13 +53,13 @@ class SendImageData {
     {
     }
 
-    const Arguments& arguments() const
+    const auto& arguments() const
     {
         return m_arguments;
     }
 
 private:
-    Arguments m_arguments;
+    std::tuple<const RefPtr<WebCore::ImageData>&> m_arguments;
```

Comment:
- `T` becomes `T&`
- `Ref<T>` becomes `const Ref<T>&`

If we instead made Ref, RefPtr, RetainPtr to be pass-by-value into the messages, they would be more natural types and there would not be a double indirection when accessing them.

In fact, the list could contain SharedMemory::Handle, UnixFileDescriptor, etc. too, which would then solve the move implementation issue..

Reply:
This PR affects decoding, so essentially how the arguments from the message are passed on to the handling method. With this added validation, in its current simplest form the method's parameter types have to mach the types of arguments, but the method still has flexibility of handling each argument through either type of references or just by value. It would be possible to special-case things like `Ref<>`, but I would prefer stricter matching.

For passing objects into the message during encoding, I would avoid special-casing in the generator and give more control for that purpose through how the message  is constructed. So essentially what #6485 tries to do, but a bit more polished, less intrusive, and incrementally adoptable.